QCflow/run_opt_neutral.py

```diff
@@ -37,3 +37,116 @@ def run_opt_neutral(mol_name, mol_smile, mol_dic, functional='B3LYP', basis_set=
     submit_slurm_job('pop_opt_n', mol_name)
     #goes back to previous directory
     os.chdir(os.path.dirname(os.getcwd()))
+
+def run_opt_neutral_set_dihedral(mol_name, mol_smile, functional='B3LYP', basis_set='6-31G*', dihedral_angle=0):
+    """
+    Submits a neutral optimization with population analysis when provided with the name of the mol
+
+    mol_name : name of oligomer
+
+    mol_smile : SMILE string of oligomer
+
+    functional : preset is B3LYP
+
+    basis_set : preset is 6-31G*
+
+    dihedral_angle : Angle to set dihedral to (preset is 0)
+    """
+    #goes into directory
+    os.chdir(f'{mol_name}')
+    #turns smiles string into rdkit object
+    mol = Chem.MolFromSmiles(mol_smile)
+    #gets rdkit estimated coordinates of dimer
+    mol3d = embed_molecule(mol)
+    #Saves the torsional scan as .csv and finds the lowest energy geometry
+    conf_geo = setting_dihedral(mol_smile, dihedral_angle)
+    #writes a guassian input file
+    write_gaussian('pop_opt_n', mol_name, mol_smile, functional, basis_set, mol3d, 0, conf_geo)
+    #writes the slurm file
+    write_slurm('pop_opt_n', mol_name)
+    #submits the slurm jon
+    submit_slurm_job('pop_opt_n', mol_name)
+    #goes back to previous directory
+    os.chdir(os.path.dirname(os.getcwd()))
+
+def staging_opt(mol_name, mol_dic, job_type='pop_opt_n', functional='B3LYP', basis_set1='3-21G', basis_set2='6-31G*', basis_set3='6-31G**'):
+    """
+    Checks if the geometry optimization as been completed at the highest desried basis set.
+    If it has then appends a dictionary showing this. If the calculations haven't been ran
+    all the way then runs them. If the calculation has failed then appends a dictionary
+    with the name of the oligomer and the basis set it failed at.
+
+    mol_name : the name of the oligomer as seen in the dictionary i.e. if melanin fragment (b) is combined
+                with organic electronic fragment (1) in the v1 position will be b_1_v1
+
+    mol_dic : dictionary of oligomers where key is the name of the oligomer and value is the SMILES string
+
+    job_type : Type of job ran e.g. pop_opt_n
+
+    functional : Preset is B3LYP
+
+    basis_set1 : Preset 3-21G* (Lowest basis set)
+
+    basis_set2 : Preset 6-31G* (Median basis set)
+
+    basis_set3 : Preset 6-31G** (Highest basis set)
```

Comment:
remove Highest basis set as there is likely no reason to go from 6-31G* to 6-31G**

Reply:
the basis set and method should be user-defined